scrapers_next/pa/committees.py

```diff
@@ -0,0 +1,171 @@
+            )
+            # main chair role
+            Chair_Member_role = (
+                CSS(
+                    "div.MemberInfoList-MemberWrapper.ChairWrapper div.ChairNameText div"
+                )
+                .match(self.root)[0]
+                .text.strip()
+            )
+            com.add_member(Chair_Member, Chair_Member_role)
+            # Democratic Chair member and or the minority chair member
+            Demo_Chair_Member = (
+                CSS("div.MemberInfoList-MemberWrapper.ChairWrapper div.ChairNameText a")
+                .match(self.root)[1]
+                .text.strip()
+            )
+            # Democratic Chair member and or the minority chair member role
+            Demo_Chair_Member_role = (
+                CSS(
+                    "div.MemberInfoList-MemberWrapper.ChairWrapper div.ChairNameText div"
+                )
+                .match(self.root)[1]
+                .text.strip()
+            )
+            com.add_member(Demo_Chair_Member, Demo_Chair_Member_role)
+        except IndexError:
+            pass
+        Majority_Members = CSS(
+            ".Widget.CteeInfo-MajorityList .MemberInfoList-MemberWrapper.Member"
+        ).match(self.root)
+        for mem in Majority_Members:
+            try:
+                major_member_name = CSS("div a").match_one(mem).text.strip()
+                major_mem_position = CSS(".position").match_one(mem).text.strip()
+            except SelectorError:
+                major_mem_position = "Member"
+            com.add_member(major_member_name, major_mem_position)
+        Minority_Members = CSS(
+            ".Widget.CteeInfo-MinorityList .MemberInfoList-MemberWrapper.Member"
+        ).match(self.root)
+        for mem in Minority_Members:
+            try:
+                minor_member_name = CSS("div a").match_one(mem).text.strip()
+                minor_mem_position = CSS(".position").match_one(mem).text.strip()
+            except SelectorError:
+                minor_mem_position = "Member"
+            com.add_member(minor_member_name, minor_mem_position)
+        return com
+
+
+class SenateCommitteeList(HtmlListPage):
+    source = "https://www.legis.state.pa.us/cfdocs/CteeInfo/StandingCommittees.cfm?CteeBody=S"
+    chamber = "upper"
+    selector = CSS("table tbody tr td:nth-child(1) a")
+
+    def process_item(self, item):
+        name = item.text_content().strip()
+        com = ScrapeCommittee(
+            name=name, classification="committee", chamber=self.chamber
+        )
+        detail_link = item.get("href")
+        com.add_source(detail_link)
+        com.add_link(detail_link, "homepage")
+        return SenateCommitteeDetail(com, source=detail_link)
+
+
+class HouseCommitteeList(HtmlListPage):
+    source = "https://www.legis.state.pa.us/cfdocs/CteeInfo/StandingCommittees.cfm?CteeBody=H"
+    chamber = "lower"
+    selector = CSS("table tbody tr td:nth-child(1) a")
+
+    def process_item(self, item):
```

Comment:
Similarly, I note that  the `process_item` method of `HouseCommitteeList` and `SenateCommitteeList` are *almost* identical. This code could be refactored out to a shared function. However, because the repeated code is brief, it is more obvious that this code is being repeated. So it's a judgement call but I personally would not bother refactoring these out to a single function.

There is something called the [DRY principle - Don't Repeat Yourself](https://thevaluable.dev/dry-principle-cost-benefit-example/). It's a good thing to remember, but as that article states, it's not an absolute. It's part of the learning process to understand when to apply it by refactoring code to avoid repeating yourself, and when a little bit of repeating is OK.